### PR TITLE
MSVC late cleanup

### PR DESCRIPTION
**Pull request by res2k**
Another PR not necessarily meant for merging, but to run CI checks.
Contains part of the changes discussed in #508.

### CODE REVIEW

File: src/init.c [hunk 1]
```diff
@@ -522,6 +527,8 @@ static void mi_process_done(void) {
   process_done = true;
 
   #if defined(_WIN32) && !defined(MI_SHARED_LIB)
+  // Explicitly clean up main thread
+  _mi_thread_done(_mi_heap_default);
```
Comment (daanx) [inline src/init.c:534]: I think this should not call _mi_thread_done as it is from the main thread. (before, line 525 prevented this by setting the fls value to null)

**Comment by daanx** [comment]
Hi @res2k, great PR! I agree with the just making padding atomic is fine; I will merge and maybe make some small adjustments and do more testing at some larger services here. The added tests are good too. Thanks so much for your in-depth investigation :-)

File: src/init.c (hunk 1 above)
Comment (res2k) [reply src/init.c:534]: But what confuses me is that you merged the PR nevertheless...
Should I make a follow-up that addresses this?

Comment (daanx) [reply src/init.c:534]: Ah, I corrected it in another later commit :-)
The latest `dev` (and `dev-slice`) should be good.

**Pull request merged**